### PR TITLE
register `tune_args()` and `tunable()` methods unconditionally

### PR DESCRIPTION
**Pull request by simonpcouch**
Similar rationale and implementation as tidymodels/workflows#193. :)

### CODE REVIEW

**Review by EmilHvitfeldt — APPROVED** [review]

File: R/tune_args.R [hunk 1]
```diff
@@ -1,6 +1,10 @@
+#' @importFrom generics tune_args
+#' @export
+generics::tune_args
 
-# Lazily registered in .onLoad()
-tune_args_model_spec <- function(object, full = FALSE, ...) {
+#' @method tune_args model_spec
```
Comment (simonpcouch) [inline R/tune_args.R:5]: For whatever reason, this one required the explicit `@method` tag to probably document as an S3 method.

File: R/tunable.R [hunk 2]
```diff
@@ -1,7 +1,13 @@
-# Lazily registered in .onLoad()
+
 # Unit tests are in extratests
 # nocov start
-tunable_model_spec <- function(x, ...) {
+
+#' @importFrom generics tunable
```
Comment (hfrick) [inline R/tunable.R:5]: this is already imported in `parsnip-package.R` so let's remove one of them. I'd remove this one here because I also think that it shouldn't be (re)exported (see https://github.com/tidymodels/workflows/pull/193#pullrequestreview-1300378027)

File: R/tune_args.R [hunk 3]
```diff
@@ -1,6 +1,10 @@
+#' @importFrom generics tune_args
```
Comment (hfrick) [inline R/tune_args.R:1]: similar comment on import/export and where it should sit

**Review by hfrick — COMMENTED** [review]
same thinking around reexporting the generics for `tune_args()` and `tunable()`

File: R/zzz.R [hunk 4]
```diff
@@ -9,45 +9,9 @@
   vctrs::s3_register("generics::tidy", "_fishnet")
   vctrs::s3_register("generics::glance", "model_fit")
   vctrs::s3_register("generics::augment", "model_fit")
-  vctrs::s3_register("generics::required_pkgs", "model_fit")
-  vctrs::s3_register("generics::required_pkgs", "model_spec")
```
Comment (hfrick) [inline R/zzz.R:13]: should this stay in this PR? given that we register it without vctrs in `R/required_pkgs.R`, it should definitely go, it's more a question of "should this be its own PR"? Up to you!

**Review by hfrick — APPROVED** [review]
Approve! 🙌

File: R/zzz.R (hunk 4 above)
Comment (simonpcouch) [reply R/zzz.R]: I'm with you that this probably ought to live in its own PR! Reverting.
